client/src/common/components/Filter.jsx

```diff
@@ -45,28 +64,17 @@ class Filter extends PureComponent {
 
   handleInputChange = event => {
     const { value } = event.target;
-    const query = _trim(value);
-
-    this.setState(() => ({ query }), this.handleFilterOptions);
-  };
-
-  handleFilterOptions = () => {
-    const { query } = this.state;
 
-    this.debouncedFilter(query);
+    this.setState(() => ({ query: value }), () => this.debouncedFilter(value));
```

Comment:
Do you need function if not using previous state?
Isn't this enough? `this.setState({ query: value }, () => this.debouncedFilter(value));`